pinot-server/src/main/java/org/apache/pinot/server/starter/helix/SegmentOnlineOfflineStateModelFactory.java

```diff
@@ -196,6 +196,8 @@ public void onBecomeDroppedFromOffline(Message message, NotificationContext cont
       String tableNameWithType = message.getResourceName();
       String segmentName = message.getPartitionName();
 
+      _instanceDataManager.deleteSegment(tableNameWithType, segmentName);
```

Comment:
This line should be inserted after line 184, not here. The segment is not available for any new queries sometime after line 184 starts execution

Reply:
Also, can we not get the table data manager directly from here? Do we need to go through the instanceDataManager? Dependency issues?

Reply:
After line 184, the segment is unavailable correct; and if it is due to a move/rebalance, we should count it as an error. We only want to suppress it when the segment has been 'deleted'.